examples/boids/src/slider.rs

```diff
@@ -68,10 +68,14 @@ impl Component for Slider {
             10f64.powi(-(p as i32))
         });
 
-        let oninput = onchange.reform(|e: InputEvent| {
-            let input: HtmlInputElement = e.target_unchecked_into();
-            input.value_as_number()
-        });
```

Comment:
Is there a good reason to remove `reform`? I think this example makes a pretty good argument for why it's convenient?

Reply:
It's explained in https://github.com/yewstack/yew/issues/2196

It mostly game down to it's existence not really mattering. It saves a couple of lines at best and imo, is hard to find. Unless you already know it's there, you end up implementing it yourself.

Reply:
That's not very convincing to me. One issue is down to lack of documentation and the other argument is like saying the `Iterator` trait doesn't need to implement `count`, `last`, `for_each`, `partition`, and many others, because ALL of them can be implemented using `fold` without increasing the line count.

To be clear, I'm not at all attached to this method. I don't think its behaviour is very intuitive to most people. I just don't see a reason to remove it here.

Reply:
I think @voidpumpkin can explain this better than I can, since it was it's removal was proposed by them

Reply:
@siku2 We had a discussion on discord regarding this, essentially it all came down to both ways to write code being the same amount of lines.
[Case #1](https://discord.com/channels/701068342760570933/703449306497024049/913050523316416542)
```rust
let Props {on_click} = props.clone();

let my_info_i_need = //etc...

let onclick = Callback::from(move |event| on_click.emit(my_info_i_need));

//html!
<div {onclick} />
```
[Case #2](https://discord.com/channels/701068342760570933/703449306497024049/913050887008702504)
```rust
let Props {on_click} = props.clone();

let my_info_i_need = //etc...

let onclick = on_click.reform(move |event| on_click.emit(my_info_i_need));

//html!
<div {onclick} />
```
Nobody provided any more reasons why have `reform`, nor any pointers to where the naming convention of `reform` comes from.
Iis not even used much either, take a look at examples.

Reply:
Your case 2 doesn't actually represent how the function is used, but regardless, for me this is less about the actual function and more about the needless breaking change. I'm more than fine with breaking changes if they make way for even just a small improvement, but in this case I just don't see it.

Reply:
@siku2 I would agree with not removing this if we were not in 0.x version.
Besides its an easy fix as its quite unlikely anyone created a method named the same, so we can just find all replace from `\s\S*.reform\(` to `Callback::from(`

But really i never seen anyone anyone creating anything with a name `reform` that is not talking about shapes. I wonder perhaps `.map()` would fit this method more?

Reply:
@hamza1311 you could even add that regex to migration guide

Reply:
@siku2 could you show how that function is supposed to be used?

Reply:
It's not that simple. reform also emitted to the original callback whereas `Callback::from` wouldn't.

Maybe I should just revert the commit and leave the method there? There's only one example (this one this thread is on) where it really matters

Reply:
@hamza1311 so it emits the reformed event ? would 
`let myclick = on_click.reform(|_| ());`
still emit `on_click` ?

Reply:
```rust
 ​    ​pub​ ​fn​ ​reform​<F, T>(​&​self​, func: F) -> Callback<T> 
 ​    ​where 
 ​        F: ​Fn​(T) -> IN + 'static, 
 ​    { 
 ​        ​let​ this ​=​ ​self​.​clone​(); 
 ​        ​let​ func ​=​ ​move​ ​|​input​|​ { 
 ​            ​let​ output ​=​ ​func​(input); 
 ​            this.​emit​(output); 
 ​        }; 
 ​        Callback​::​from​(func) 
 ​    }
```

It'll call the given function and emit the value returned by it to the original callback.

Reply:
@hamza1311 I guess we can leave it then as it does serve a purpose. But we really need to document it.

Reply:
> @siku2 I would agree with not removing this if we were not in 0.x version.

I wholeheartedly agree with this, but it's not a good excuse for removing a method out of the blue. If it was somehow inefficient to use this method or if the Callback internals were to change and this method no longer made sense then yeah, absolutely, remove it. Even keeping the API slim and clean would be a valid reason, but I don't think that applies here since the method has a clear and well defined purpose. As it stands we would only really run the risk of annoying the users (if there even are any :P) of this function.

Reply:
@siku2 yeah sorry for all this, we truly missed the mark during that discord discussion. I agree that we should keep it.

Reply:
I reverted the commit. I left the changes in `function_todomvc` example as it demonstrate the fact that closures can be passed to event handlers, which is important in function components.